edit-release.php

```diff
@@ -0,0 +1,416 @@
+			(SELECT created FROM `release` WHERE modid = `mod`.modid ORDER BY created DESC LIMIT 1),
+			`mod`.created
+		)
+		WHERE modid = ?;
+	', [$targetMod['modid']]);
+
+	$con->completeTrans();
+
+	forceRedirect(formatModPath($targetMod).'#tab-files');
+	exit();
+}
+
+
+//
+// Prepare data for display
+//
+
+if($existingRelease) {
+	$files = $con->getAll("
+		SELECT *, CONCAT(ST_X(imagesize), 'x', ST_Y(imagesize)) AS imagesize
+		FROM file
+		WHERE assetid = ?
+	", [$existingRelease['assetid']]);
+
+	$compatibleGameVersions = $con->getCol('SELECT gameVersion FROM ModReleaseCompatibleGameVersions WHERE releaseId = ?', $existingRelease['releaseid']);
+	$existingRelease['compatibleGameVersions'] = array_flip(array_map('intval', $compatibleGameVersions));
+}
+else {
+	// hovering files
+	$files = $con->getAll("
+		SELECT *, file.fileid, CONCAT(ST_X(imagesize), 'x', ST_Y(imagesize)) AS imagesize, mpr.detectedmodidstr, mpr.detectedmodversion
+		FROM file
+		LEFT JOIN modpeek_result mpr ON mpr.fileid = file.fileid
+		WHERE assetid IS NULL AND assettypeid = 2 AND userid = ?
+	", [$user['userid']]);
+}
+
+foreach($files as &$file) {
+	$file["created"] = date("M jS Y, H:i:s", strtotime($file["created"]));
+
+	$file["ext"] = substr($file["filename"], strrpos($file["filename"], ".")+1); // no clue why pathinfo doesnt work here
+	$file["url"] = maybeFormatDownloadTrackingUrlDependingOnFileExt($file);
+}
+unset($file);
+
+
+
+$allGameVersions = $con->getAll('SELECT version FROM GameVersions ORDER BY version DESC');
+foreach($allGameVersions as &$gameVersion) {
+	$gameVersion['version'] = intval($gameVersion['version']);
+	$gameVersion['name'] = formatSemanticVersion($gameVersion['version']);
+}
+unset($gameVersion);
+
+
+$assetChangelog = $existingRelease ? $con->getAll('
+	SELECT changelog.*, user.name as username
+	FROM changelog
+	JOIN user ON changelog.userid = user.userid
+	WHERE changelog.assetid = ?
+	ORDER BY created DESC
+	LIMIT 20
+', [$existingRelease['assetid']]) : [];
+
+
+
+if(!$existingRelease) {
+	$existingRelease = [
+		'assetid'    => 0,
+		'text'       => $_POST['text'] ?? '',
+		'numsaved'   => 0,
+		'compatibleGameVersions' => empty($_POST['cgvs']) ? [] : array_flip(array_filter(array_map('compileSemanticVersion', $_POST['cgvs']))),
+	];
+
+	if($targetMod['type'] === 'mod') {
+		$existingRelease['modidstr']   = $files ? $files[0]['detectedmodidstr'] : '';
+		$existingRelease['modversion'] = $files ? $files[0]['detectedmodversion'] : '';
+	}
+	else {
+		$existingRelease['modidstr']   = '';
+		$existingRelease['modversion'] = $_POST['modversion'] ?? '';
+	}
+}
+else {
+	$existingRelease['modversion'] = formatSemanticVersion(intval($existingRelease['modversion']));
+}
+
+$view->assign('allGameVersions', $allGameVersions);
+
+$view->assign('mod', $targetMod);
+$view->assign('release', $existingRelease);
+$view->assign('files', $files);
+
+$view->assign('assetChangelog', $assetChangelog);
+
+$view->display('edit-release');
```

Comment:
not sure about this rewrite. this seems less organized. The old code at least was separated into load(), save(), delete(), "errorHandling()"

Reply:
I can split it up into functions (which would have to happen either way if we want to expose this on the api), but i didn't really see the point in doing so as all of the code is linar, none of the functions would be called more than once and i would have to pass quite a few arguments back and forth between them.

The main issue with the old version was its specific structure, i don't necessarily have a problem with splitting it into functions. I can do so and move the logic into lib/ so it can be reused.